src/Compilers/CSharp/Portable/Symbols/Source/SourceNamespaceSymbol.cs

```diff
@@ -228,8 +228,8 @@ internal override NamespaceExtent Extent
                     // NOTE: the following is not cancellable.  Once we've set the
                     // members, we *must* do the following to make sure we're in a consistent state.
                     this.DeclaringCompilation.DeclarationDiagnostics.AddRange(diagnostics);
-
                     RegisterDeclaredCorTypes();
+                    DeclaringCompilation.SymbolDeclaredEvent(this);
```

Comment:
Sure, I will document how the placement of this call relaxes the ordering constraints. Though actually it *could* be earlier.